### PR TITLE
LLVM split and patch cleanup

### PR DESCRIPTION
**Pull request by mgorny**
This is still work-in-progress:

- [x] sys-devel/llvm
- [x] sys-devel/clang (+ clang-tools-extra)
- [x] sys-devel/compiler-rt

Known bugs:
- [x] llvm -- clean up unused patches
- [x] clang -- docs ain't built
- [ ] compiler-rt -- tests ain't happy with out-of-llvm build

Also requires user patches:
- llvm: 
  1. https://reviews.llvm.org/D23742
  2. https://reviews.llvm.org/D23743
- clang:
  1. https://reviews.llvm.org/D23744
  2. https://reviews.llvm.org/D23745
  3. https://reviews.llvm.org/D23752
  4. https://reviews.llvm.org/D23754
  5. https://reviews.llvm.org/D23758

### CODE REVIEW

**Comment by mgorny** [comment]
@williamh, @austin987, @zzlei -- this is what I'm working on right now. It looks much better than it did in the past, so hopefully I'll be able to push it for -9999. This also involves llvm-config fix that should keep rust happy.

**Comment by zzlei** [comment]
I'm also having trouble with the test cases of libc++ and libc++abi; look forward to your solution to this :)

File: sys-devel/clang/clang-9999.ebuild [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+RDEPEND="
+	~sys-devel/llvm-${PV}:=[debug=,multitarget?,video_cards_radeon?,${MULTILIB_USEDEP}]
+	static-analyzer? ( dev-lang/perl:* )
+	xml? ( dev-libs/libxml2:2=[${MULTILIB_USEDEP}] )
+	!<sys-devel/llvm-${PV}
+	${PYTHON_DEPS}"
+# configparser-3.2 breaks the build (3.3 or none at all are fine)
+DEPEND="${RDEPEND}
+	xml? ( virtual/pkgconfig )
```
Comment (austin987) [inline sys-devel/clang/clang-9999.ebuild]: doc should come before xml

**Comment by austin987** [comment]
Look mostly okay at a quick glance, I'll try to test the ebuilds this week when I have time.

File: sys-devel/clang/clang-9999.ebuild (hunk 1 above)
Comment (mgorny) [inline sys-devel/clang/clang-9999.ebuild]: fixed.

**Comment by mgorny** [comment]
Added necessary (upstreamed) patches and grouped patches by latest version applicable, so we can easily wipe old patches along with old ebuilds. Let's wait for one more QA run and merge afterwards.

**Pull request closed without merging**